pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -515,6 +516,13 @@ func (c *Cluster) makeDeployment(osdProps osdProperties, osd OSDInfo, provisionC
 		dataPath = activateOSDMountPath + osdID
 	}
 
+	// Inject extra envrinoment variables.
+	if extraEnvVarsValue, err := k8sutil.GetOperatorSetting(context.TODO(), c.context.Clientset, controller.OperatorSettingConfigMapName, "OSD_EXTRA_ENV_VARS", ""); err != nil {
```

Comment:
```suggestion
	if extraEnvVarsValue, err := k8sutil.GetOperatorSetting(c.clusterInfo.Context, c.context.Clientset, controller.OperatorSettingConfigMapName, "OSD_EXTRA_ENV_VARS", ""); err != nil {
```

No need for a new context.